source/print.c

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 10xEngineers
+
+#include "../common/print.h"
+
+/**
+ * @brief Function to print results of different types
+ * 
+ * This function prints the results from an array of integers of various types
+ * (int8_t, int16_t, int32_t, int64_t, uint8_t, uint16_t, uint32_t, uint64_t)
+ * with the specified number of elements ('size').
+ * 
+ * @param results Pointer to the array containing the results to be printed
+ * @param size Number of elements in the array
+ * @param type Type of elements in the results array i.e "int8_t", "uint8_t"
+ */
+void print_results(void *results, int size, const char *type) {
+    int i;
+    if (strcmp(type, "int8_t") == 0) {  // int8_t
```

Comment:
use switch case.

Reply:
The type should be macro'ed for the int size.

Reply:
done